src/docgen/view/documentation.ts

```diff
@@ -1,121 +1,227 @@
  */
-export interface DocumentationOptions {
+export interface RenderOptions {
+
   /**
-   * The assembly to generate from.
+   * Include a generated api reference in the documentation.
+   *
+   * @default true
    */
-  readonly assembly: reflect.Assembly;
+  readonly apiReference?: boolean;
 
   /**
-   * Which language to generate docs for.
-   */
-  readonly language: string;
+    * Include the user defined README.md in the documentation.
+    *
+    * @default true
+    */
+  readonly readme?: boolean;
 
   /**
-   * Include a generated api reference in the documentation.
+    * Generate documentation only for a specific submodule.
+    *
+    * @default - Documentation is generated for the root module only.
+    */
+  readonly submodule?: string;
+
+}
+
+/**
+ * Options for creating a `Documentation` object.
+ */
+export interface DocumentationOptions {
+
+  /**
+   * Which language to generate docs for.
    *
-   * @default true
+   * @default 'ts'
    */
-  readonly apiReference?: boolean;
+  readonly language?: string;
 
   /**
-   * Include the user defined README.md in the documentation.
+   * Whether to ignore missing fixture files that will prevent transliterating
+   * some code snippet examples.
    *
    * @default true
    */
-  readonly readme?: boolean;
+  readonly loose?: boolean;
+}
+
+/**
+ * Options for creating a `Documentation` object using the `fromLocalPackage` function.
+ */
+export interface FromLocalPackageDocumentationOptions extends DocumentationOptions {
 
   /**
-   * Generate documentation only for a specific submodule.
+   * A local directory containing jsii assembly files that will
+   * comprise the type-system.
    *
-   * @default - Documentation is generated for all submodules.
+   * @default - the root package directory will be used.
    */
-  readonly submoduleName?: string;
+  readonly assembliesDir?: string;
 }
 
 /**
  * Render documentation pages for a jsii library.
  */
 export class Documentation {
-  private readonly assembly: reflect.Assembly;
-  private readonly submodule?: reflect.Submodule;
-
-  private readonly includeReadme: boolean;
-  private readonly includeApiReference: boolean;
-
-  private readonly transpile: Transpile;
-
-  constructor(options: DocumentationOptions) {
-    this.assembly = options.assembly;
-
-    switch (options.language) {
-      case 'python':
-        this.transpile = new PythonTranspile();
-        break;
-      case 'ts':
-        this.transpile = new TypeScriptTranspile();
-        break;
-      default:
-        throw new Error(
-          `Generating documentation for ${options.language} is not supported`,
-        );
-    }
 
-    if (options.submoduleName) {
-      this.submodule = this.findSubmodule(options.submoduleName);
+  /**
+   * Create a `Documentation` object for a remote package available in the npm registry.
+   */
+  public static async forRemotePackage(name: string, version: string, options?: DocumentationOptions): Promise<Documentation> {
```

Comment:
Maybe this should be called `fromNpm()` and `fromDirectory()` ("remote" is not that clear).

Reply:
There are also interesting questions about how the package is fetched... It might be relevant to rename this `forRegistryPackage`, especially if this will transparently use system/user-wide configuration for which registry is being used...

Reply:
I like `forRegistryPackage` 👍  - later on we can add a registry URL option for it so that we won't have to use system-wide configurations.

Reply:
Nit: I think the "from" convention for factories is more robust. "For" works for documentation but not for other things. 

For example, "documentation for directory" doesn't work while "Documentation [object] from directory" works.

But this is just me being me

Reply:
I agree that `forDirectory` doesn't work, but I also think its because of the "directory" - it doesn't have any semantic meaning. So the solution here could also be to come up with a proper semantic name for it. 

It is now `forProject` - which signals that its meant to be used on locally built projects with a `package.json` in its root directory.